### PR TITLE
Graduate SelfLink deprecation to GA

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/enhancements/issues/1164

/assign @liggitt

### CODE REVIEW

**Review by sftim — DISMISSED** [review]
Thanks @wojtek-t 

Let's document which release deprecated `selfLink`, and edit the feature gate description to make that clear. The current text is a bit off because a literal reading implies the end user has control over whether the field is deprecated.

Additionally, we should describe the effect of this feature gate in terms of the API and its fields (camelCase), rather than the Golang code for our reference implementation (`ListMeta`, etc).

For example:
```diff
- - `RemoveSelfLink`: Deprecates and removes `selfLink` from ObjectMeta and-
-   ListMeta.
+ - `RemoveSelfLink`: Sets the `.metadata.selfLink` field to blank (empty string) for all
+   objects and collections. This field has been deprecated since the Kubernetes v1.xx
+   release. When this feature is enabled, the `.metadata.selfLink` field remains part of
+   the Kubernetes API, but is always unset.
```

**Comment by wojtek-t** [comment]
@sftim - thanks; PTAL

File: content/en/docs/reference/kubernetes-api/common-definitions/list-meta.md [hunk 1]
```diff
@@ -44,11 +44,5 @@ ListMeta describes metadata that synthetic resources must have, including lists
...
 - **selfLink** (string)
 
-  selfLink is a URL representing this object. Populated by the system. Read-only.
-  
-  DEPRECATED Kubernetes will stop propagating this field in 1.20 release and the field is planned to be removed in 1.21 release.
-
-
-
-
+  SelfLink is DEPRECATED read-only field that is no longer populated by the system.
```
Comment (sftim) [inline content/en/docs/reference/kubernetes-api/common-definitions/list-meta.md:47]: nit: I might call this an unused and / or legacy field; “deprecated” means “you should stop using this” (ahead of future removal, etc), whereas the truth now is that it already gone. The field is retained to meet API compatibility promises but the only field value is `""`.

**Review by sftim — COMMENTED** [review]
`content/en/docs/reference/command-line-tools-reference/feature-gates.md` LGTM

We'd usually _omit_ the other changes (which are to generated files) and let the docs generation phase for the v1.24 release take care of those details.

**Comment by wojtek-t** [comment]
@sftim - thanks for comments; PTAL

File: content/en/docs/reference/kubernetes-api/common-definitions/list-meta.md [hunk 2]
```diff
@@ -44,11 +44,5 @@ ListMeta describes metadata that synthetic resources must have, including lists
...
 - **selfLink** (string)
 
-  selfLink is a URL representing this object. Populated by the system. Read-only.
-  
-  DEPRECATED Kubernetes will stop propagating this field in 1.20 release and the field is planned to be removed in 1.21 release.
-
-
-
-
+  SelfLink is a legacy read-only field that is no longer populated by the system.
```
Comment (sftim) [inline content/en/docs/reference/kubernetes-api/common-definitions/list-meta.md:47]: This change needs to happen [upstream](https://github.com/kubernetes/kubernetes/blob/2ccdd4573b6ae500b96e77638752de8f2f329fe4/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L61) first.

Comment (wojtek-t) [reply content/en/docs/reference/kubernetes-api/common-definitions/list-meta.md:47]: I opened https://github.com/kubernetes/kubernetes/pull/108163 for it.

Comment (wojtek-t) [reply content/en/docs/reference/kubernetes-api/common-definitions/list-meta.md:47]: This is now merged in upstream.

**Comment by wojtek-t** [comment]
@sftim - PTAL

**Comment by tengqm** [comment]
@wojtek-t Please remove changes to `content/en/docs/reference/kubernetes-api/common-definitions/` because those files are auto-generated. They would be taken care of when 1.24 is released.

Other than this, the PR LGTM.

**Comment by wojtek-t** [comment]
@tengqm - thanks; PTAL

**Comment by tengqm** [comment]
/lgtm

Defer approval to release maester.

**Pull request merged**